### DISCUSSION TITLE
Lets be Cranky

### DISCUSSION THREAD

**Discussion by dbuch** [nushell/nushell#7183] [Ideas]
Clippy is quite useful and is encuraged to be used. But it can by cumbersome to workwith in a workspace setup.
I came across cargo-cranky: https://github.com/ericseppanen/cargo-cranky

Perhaps we should use this? Egui has an example configuration here: https://github.com/emilk/egui/blob/master/Cranky.toml

**Comment by rgwood** [comment]
Thanks for the suggestion. As I understand it, the benefit would be that the flags we currently pass into Clippy  (`-- -D warnings -D clippy::unwrap_used -A clippy::needless_collect`) could live in a TOML file instead?

I guess we'd have to [install Cranky](https://github.com/emilk/egui/blob/8671aa26e1f65340c41f6b2c6f3ef2d6d80a6567/.github/workflows/rust.yml#L34) to make this work. The benefit doesn't quite seem worth the cost, but perhaps I'm missing something.
